components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/artifact/SAMLSSOSoapMessageService.java

```diff
@@ -0,0 +1,141 @@
+import org.opensaml.xml.XMLObjectBuilderFactory;
+import org.wso2.carbon.identity.application.authenticator.samlsso.exception.ArtifactResolutionException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.UnknownHostException;
+
+/**
+ * This class is used for handling SAML SOAP Binding
+ */
+public class SAMLSSOSoapMessageService {
+    private static Log log = LogFactory.getLog(SAMLSSOSoapMessageService.class);
+    private static final String CONTENT_TYPE = "text/xml; charset=utf-8";
+
+    /**
+     * Build a SOAP Message.
+     *
+     * @param samlMessage SAMLObject.
+     * @return Envelope soap envelope
+     */
+    public Envelope buildSOAPMessage(SAMLObject samlMessage) {
+
+        XMLObjectBuilderFactory builderFactory = Configuration.getBuilderFactory();
+
+        SOAPObjectBuilder<Envelope> envBuilder = (SOAPObjectBuilder<Envelope>) builderFactory.getBuilder(
+                Envelope.DEFAULT_ELEMENT_NAME);
+        Envelope envelope = envBuilder.buildObject();
+
+        if(log.isDebugEnabled()) {
+            log.debug("Adding SAML message to the SOAP message's body");
+        }
+        SOAPObjectBuilder<Body> bodyBuilder = (SOAPObjectBuilder<Body>) builderFactory.getBuilder(
+                Body.DEFAULT_ELEMENT_NAME);
+        Body body = bodyBuilder.buildObject();
+        body.getUnknownXMLObjects().add(samlMessage);
+        envelope.setBody(body);
+        return envelope;
+    }
+
+    /**
+     * Send SOAP message
+     *
+     * @param message message that needs to be send
+     * @param url url that the artifact resolve request should be sent
+     * @return response of invoking artifact resolve endpoint
+     * @throws ArtifactResolutionException
+     */
+    public String sendSOAP(String message, String url) throws ArtifactResolutionException {
+
+        StringBuilder soapResponse = new StringBuilder();
+        try {
+            HttpPost httpPost = new HttpPost(url);
+            setRequestProperties(url, message, httpPost);
+            DefaultHttpClient httpClient = new DefaultHttpClient();
+            HttpResponse httpResponse = httpClient.execute(httpPost);
+
+            int responseCode = httpResponse.getStatusLine().getStatusCode();
+            if (responseCode != 200) {
+                throw new ArtifactResolutionException("Problem in communicating with: " + url + ". Received response: "
+                        + responseCode);
+            } else {
+                log.info("Successful response from the URL: " + url);
+                soapResponse.append(getResponseBody(httpResponse));
+            }
+        }
+        catch (UnknownHostException eUH) {
+            throw new ArtifactResolutionException("Unknown targeted host: " + url , eUH);
```

Comment:
I would say referring the exception object as 'e' is better as that's the common practice that we have used